src/main/java/randoop/generation/ForwardGenerator.java

```diff
@@ -235,6 +236,14 @@ private void initializeRuntimePrimitivesSeen() {
 
     eSeq.execute(executionVisitor, checkGenerator);
 
+    // Dynamic type casting permits calling methods that do not exist on the declared type.
+    if (GenInputsAbstract.cast_to_run_time_type && eSeq.isNormalExecution()) {
+      castToRunTimeType(eSeq);
+      // Re-execute the sequence after applying dynamic type casting.
```

Comment:
Please only re-execute the sequence if necessary -- that is, if `castToRunTimeType` *needed* to insert a cast because it changed the variable type.